stackable-base/Dockerfile

```diff
@@ -49,63 +53,86 @@ SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
-      # cpe:2.3:a:gnu:tar:-:*:*:*:*:*:*:* \
-      tar \
-      # Added only temporarily, removed again below
-      shadow-utils && \
-    groupadd --gid 1000 --system stackable && \
-    useradd --gid stackable --uid 1000 --system stackable -d /stackable && \
-    mkdir /stackable && \
-    chown stackable:stackable /stackable && \
-    microdnf remove shadow-utils && \
-    microdnf clean all && \
-    echo "alias ll='ls -alF --color=auto'" >> /stackable/.bashrc && \
-    echo "alias ls='ls --color=auto'" >> /stackable/.bashrc && \
-    echo "alias ..='cd ..'" >> /stackable/.bashrc && \
-    echo "export PS1='\u@\[\e[36m\]\H\[\e[m\] \[\e[32m\]\$(pwd)\[\e[m\] \\$ '" >> /stackable/.bashrc && \
-    echo -e "if [ -f ~/.bashrc ]; then\n\tsource ~/.bashrc\nfi" >> /stackable/.profile && \
-    chown stackable:stackable /stackable/.bashrc && \
-    chown stackable:stackable /stackable/.profile
+
+# echo won't expand escape sequences. Consider printf.
+# https://github.com/koalaman/shellcheck/wiki/SC2028
+# Reason: This is complaining about the complicated PS1 statement.
+# While it seems like black it seems to work as intended so I'm not going to touch it!
+# hadolint ignore=SC2028
+RUN <<EOF
+microdnf update
+
+# **iputils**
+# To make debugging easier, includes things like ping
+# Added 2024-03: We cannot find any vulnerabilities in the past years
+# https://github.com/iputils/iputils
+#
+# **less**
+# To make debugging easier
+# Added 2024-03: less has seen three vulnerabilities between 2004 and 2022 which is a risk we're willing to accept for the added convenience
+# https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&search_type=all&isCpeNameSearch=false&cpe_vendor=cpe%3A%2F%3Agnu&cpe_product=cpe%3A%2F%3A%3Aless
+# cpe:2.3:a:gnu:less:*:*:*:*:*:*:*:*
+#
+# **nano**
+# To make debugging and changing things easier
+# Added 2024-03: We checked and it has not seen any vulnerabilities since 2010 (as of 2024-03) we decided to accept it into our base image
+# https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&query=cpe%3A2.3%3Aa%3Agnu%3Anano&search_type=all&isCpeNameSearch=false
+# cpe:2.3:a:gnu:nano:*:*:*:*:*:*:*:*
+#
+# **tar**
+# To enable kubectl cp
+# Added 2024-03: We checked and it has seen eight vulnerabilities since 2001, mostly minor and it's not in executable path so we decided to accept the risk
+# https://nvd.nist.gov/vuln/search/results?adv_search=true&isCpeNameSearch=true&query=cpe%3A2.3%3Aa%3Agnu%3Atar%3A-%3A*%3A*%3A*%3A*%3A*%3A*%3A*
+# cpe:2.3:a:gnu:tar:-:*:*:*:*:*:*:*
+microdnf install \
+  iputils \
+  less \
+  nano \
+  tar
+
+# Added only temporarily to create the user and group, removed again below
+microdnf install shadow-utils
+groupadd --gid 1000 --system stackable
+useradd --gid stackable --uid 1000 --system stackable -d /stackable
+
+mkdir /stackable
+chown stackable:stackable /stackable
```

Comment:
I always wondered why we don't properly assign `/stackable` as home dir to the `useradd` command and let it do the right thing.

Reply:
I can try to look at that in a follow-up PR. I wanted this to be purely "formatting" changes.

Reply:
It looks like we already do and these two lines might actually be redundant. I'll check!